File: src/android/src/main/kotlin/com/circuids/boat/pipeline/MutableAudioFrame.kt
package com.circuids.boat.pipeline

/**
 * Pre-allocated mutable frame reused across pipeline iterations.
 * Zero allocation per frame.
 */
class MutableAudioFrame(val pcm: ByteArray) {
    var sequenceNumber: Long = 0
    var timestampNanos: Long = 0
    var sampleRate: Int = 16000
    var channelCount: Int = 1
    var speechActive: Boolean = false
    var speechConfidence: Float = 0f
    var processingTimeNanos: Long = 0
    var dropped: Boolean = false
    var validBytes: Int = pcm.size

    fun reset() {
        sequenceNumber = 0
        timestampNanos = 0
        speechActive = false
        speechConfidence = 0f
        processingTimeNanos = 0
        dropped = false
        validBytes = pcm.size
    }
}

File: src/ios/Classes/Pipeline/MutableAudioFrame.swift
import Foundation

/// Pre-allocated mutable frame reused across pipeline iterations.
final class MutableAudioFrame {
    var pcm: Data
    var sequenceNumber: Int64 = 0
    var timestampNanos: Int64 = 0
    var sampleRate: Int = 16000
    var channelCount: Int = 1
    var speechActive: Bool = false
    var speechConfidence: Float = 0
    var processingTimeNanos: Int64 = 0
    var dropped: Bool = false
    var validBytes: Int

    init(pcm: Data) {
        self.pcm = pcm
        self.validBytes = pcm.count
    }

    func reset() {
        sequenceNumber = 0
        timestampNanos = 0
        speechActive = false
        speechConfidence = 0
        processingTimeNanos = 0
        dropped = false
        validBytes = pcm.count
    }
}
